roles/pulpcore_repositories/tasks/main.yml

```diff
@@ -24,4 +24,16 @@
     creates: /etc/dnf/modules.d/pulpcore.module
   when:
     - pulpcore_repositories_version is version('3.16', '>=')
+    - pulpcore_repositories_version is version('3.39', '<')
+    - ansible_distribution_major_version == '8'
+
+- name: 'Disable pulpcore:el8 module'
+  command: dnf module disable -y pulpcore:el8
+  # can't use the `dnf` module for modules without a default stream
+  # https://github.com/ansible/ansible/issues/56504
+  # https://github.com/ansible/ansible/issues/64852
+  args:
+    creates: /etc/dnf/modules.d/pulpcore.module
```

Comment:
I don't think this is right. The file will already be there if it is enabled and instead it will actually skip this step if it's enabled (which is true for upgrades).

Just for background:
```console
# dnf module enable pulpcore:el8
# ls /etc/dnf/modules.d
pulpcore.module  python39.module
# dnf module disable pulpcore
# ls /etc/dnf/modules.d
pulpcore.module  python39.module
# cat /etc/dnf/modules.d/pulpcore.module
[pulpcore]
name=pulpcore
stream=
profiles=
state=disabled
```

Note that the enable part also enabled python39 but disable doesn't disable python. It looks like ansible-core has rebased to python3.11 (al least in CentOS Stream 8), but worth considering in the release notes.

Reply:
@evgeni any thoughts on using this for idempotency?

```yaml
- name: 'Disable pulpcore:el8 module'
  command: dnf module disable -y pulpcore:el8
  args:
    removes: /etc/dnf/modules.d/pulpcore.module

- name: Delete pulpcore module
  file:
    path: /etc/dnf/modules.d/pulpcore.module
    state: absent
```

Reply:
This looks reasonable. The `removes` on the disable command is a "lie", but then you fix it up and things should be fine.

OTOH, we could also *just* remove the file, that's the only thing that carries the info about the module state.

Reply:
I went with remove only and updated the PR